tests/update_only_fields/fields.py

```diff
@@ -0,0 +1,43 @@
+from django.db.models import Field
+
+
+class CompoundField(Field):
+    '''Non-concrete field that takes two concrete fields
+    and returns a tuple that combines their values'''
+
+    def __init__(
+        self,
+        a_field="field_a",
+        b_field="field_b",
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.a_field = a_field
+        self.b_field = b_field
+
+    def __get__(self, instance, cls=None):
+        if instance is None:
+            return self
+
+        a = getattr(instance, self.a_field)
+        b = getattr(instance, self.b_field)
+        return (a, b)
+
+    def __set__(self, instance, value):
+        if value:
+            a, b = value
+        else:
+            a, b = self.get_default()
+        setattr(instance, self.a_field, a)
+        setattr(instance, self.b_field, b)
+
+    def contribute_to_class(self, cls, name, **kwargs):
+        super().contribute_to_class(cls, name, **kwargs)
+        setattr(cls, name, self)
+
+    def get_attname_column(self):
+        attname, _ = super().get_attname_column()
+        return attname, None
+
+    def get_default(self):
+        return ("", "")
```

Comment:
We don't need a full implementation of a compound field for this test, I would add a simple non-concrete field directly in models, e.g.

```python
class NonConcreteField(models.IntegerField):
    def db_type(self, connection):
        return None

    def get_attname_column(self):
        attname, _ = super().get_attname_column()
        return attname, None

class Profile(models.Model):
    ...
    non_concrete = NonConcreteField()
```